src/dune_memory/dune_memory.ml

```diff
@@ -155,8 +173,10 @@ module Memory = struct
         repo metadata
     in
     let promote (path, expected_hash) =
-      Log.infof "promote %s" (Path.to_string path);
-      let stat = Unix.lstat (Path.to_string path) in
+      make_path memory path
+      >>= fun abs_path ->
```

Comment:
Can you use `let*` and `let+` instead of `>>=` and `>>|` please?

Reply:
Sure ; However I have another branch on top of this that implements deduplication, I'd rather change the whole codebase to `let*` / `let+` in one go after it's merged to avoid conflicts.

Reply:
Okay, that's fine with me.